apps/examples/messaging_sample/messaging_main.c

```diff
@@ -0,0 +1,226 @@
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <sched.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <messaging/messaging.h>
+
+#define TEST1_PORT "test1_port"
+#define TEST1_DATA "TEST1"
+
+#define TEST2_PORT "test2_port"
+#define TEST2_DATA "TEST2"
+#define TEST2_REPLY "REPLY_MSG"
+
+#define MSG_PRIO 10
+#define TASK_PRIO 100
+#define STACKSIZE 2048
+#define BUFFER_SIZE 100
+
+static void recv_callback(void *data, void *recv_data)
+{
+	printf("Success to receive with non-block mode in recv callback. : %s\n", (char *)recv_data);
+}
+
+int async_send(int argc, FAR char *argv[])
+{
+	int ret;
+	msg_send_param_t param;
+	param.priority = MSG_PRIO;
+	param.type = MSG_SEND_TYPE_ASYNC;
+	param.msg = TEST1_DATA;
+	param.msglen = sizeof(TEST1_DATA);
+
+	printf("- Send %s data with async mode.\n", TEST1_DATA);
+	ret = messaging_send(TEST1_PORT, &param, NULL);
+	if (ret != OK) {
+		printf("Fail to async send message.\n");
+	}
+
+	return 0;
+}
+
+int sync_send(int argc, FAR char *argv[])
+{
+	int ret;
+	msg_send_param_t send_param;
+	msg_reply_param_t reply_param;
+	send_param.priority = MSG_PRIO;
+	send_param.type = MSG_SEND_TYPE_SYNC;
+	send_param.msg = TEST2_DATA;
+	send_param.msglen = sizeof(TEST2_DATA);
+	reply_param.msg = (char *)malloc(BUFFER_SIZE);
+
+	printf("- Send %s data with sync mode.\n", TEST2_DATA);
+	ret = messaging_send(TEST2_PORT, &send_param, &reply_param);
+	if (ret != OK) {
+		printf("Fail to sync send message.\n");
+	}
+
+	printf("Success to receive reply from receiver. : %s\n", (char *)reply_param.msg);
+	free(reply_param.msg);
+
+	return 0;
+}
+
+int nonblock_recv(int argc, FAR char *argv[])
+{
+	int ret;
+	msg_recv_param_t param;
+
+	param.type = MSG_RECV_TYPE_NONBLOCK;
+	param.cb_func = recv_callback;
+	param.msg = (char *)malloc(BUFFER_SIZE);
+	param.msglen = BUFFER_SIZE;
+
+	ret = messaging_recv(TEST1_PORT, &param);
+	if (ret != OK) {
+		printf("Fail to receive with non-block mode.\n");
+		free(param.msg);
+	}
+
+	return OK;
+}
+
+int block_recv(int argc, FAR char *argv[])
+{
+	int ret;
+	msg_recv_param_t recv_param;
+	msg_reply_param_t reply_param;
+
+	recv_param.type = MSG_RECV_TYPE_BLOCK;
+	recv_param.msg = (char *)malloc(BUFFER_SIZE);
+	if (recv_param.msg == NULL) {
+		printf("Fail to receive, because out of memory.\n");
+	}
+	recv_param.msglen = BUFFER_SIZE;
+
+	ret = messaging_recv(TEST2_PORT, &recv_param);
+	if (ret != OK) {
+		printf("Fail to receive with block mode.\n");
+		free(recv_param.msg);
+	}
+
+	printf("Success to receive with block mode in receiver. : %s\n", (char *)recv_param.msg);
+	printf("- Reply to sender %s data.\n", TEST2_REPLY);
+
+	reply_param.msg = TEST2_REPLY;
+	reply_param.msglen = sizeof(TEST2_REPLY);
+	ret = messaging_send_reply(TEST2_PORT, &reply_param);
+	if (ret != OK) {
+		printf("Fail to reply.\n");
+		free(recv_param.msg);
+	}
+
+	free(recv_param.msg);
+	return OK;
+}
+
+void async_nonblock_messaging_test(void)
```

Comment:
This tc should use 2 thread properly, so if it couldn't create any of 2 thread, test must be stopeed

Reply:
I modified. Thanks.